packages/@aws-cdk/cicd/README.md

```diff
@@ -0,0 +1,112 @@
+## Continuous Integration / Continuous Delivery for CDK Applications
+This library includes a *CodePipeline* action for deploying AWS CDK Applications.
+
+### Getting Started
+In order to add the `PipelineDeployStackAction` to your *CodePipeline*, you need to have a *CodePipeline* artifact that
+contains the result of invoking `cdk synth -o <dir>` on your *CDK App*. You can for example achieve this using a
+*CodeBuild* project.
+
+The example below defines a *CDK App* that contains 3 stacks:
+* `CodePipelineStack` manages the *CodePipeline* resources, and self-updates before deploying any other stack
+* `ServiceStackA` and `ServiceStackB` are service infrastructure stacks, and need to be deployed in this order
+
+```
+  ┏━━━━━━━━━━━━━━━━┓  ┏━━━━━━━━━━━━━━━━┓  ┏━━━━━━━━━━━━━━━━┓  ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+  ┃     Source     ┃  ┃     Build      ┃  ┃  Self-Update   ┃  ┃            Deploy             ┃
+  ┃                ┃  ┃                ┃  ┃                ┃  ┃                               ┃
+  ┃ ┌────────────┐ ┃  ┃ ┌────────────┐ ┃  ┃ ┌────────────┐ ┃  ┃ ┌────────────┐ ┌────────────┐ ┃
+  ┃ │   GitHub   ┣━╋━━╋━▶ CodeBuild  ┣━╋━━╋━▶Deploy Stack┣━╋━━╋━▶Deploy Stack┣━▶Deploy Stack│ ┃
+  ┃ └────────────┘ ┃  ┃ └────────────┘ ┃  ┃ └────────────┘ ┃  ┃ └────────────┘ └────────────┘ ┃
+  ┗━━━━━━━━━━━━━━━━┛  ┗━━━━━━━━━━━━━━━━┛  ┗━━━━━━━━━━━━━━━━┛  ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+```
+
+#### `index.ts`
+```ts
+import codebuild = require('@aws-cdk/aws-codebuild');
+import codepipeline = require('@aws-cdk/aws-codepipeline');
+import cdk = require('@aws-cdk/cdk');
+import cicd = require('@aws-cdk/cicd');
+
+const app = new cdk.App();
+
+// We define a stack that contains the CodePipeline
+const pipelineStack = new cdk.Stack(app, 'CodePipelineStack');
+const pipeline = new codepipeline.Pipeline(pipelineStack, 'CodePipeline', {
+  restartExecutionOnUpdate: true,
```

Comment:
Explain why this is important. Can we make this the default or is it unsafe?

Reply:
Added some comments. With respects to making this the default... If we do, that's opinionated & differs from the defaults you get from console & API experience...